ksql-cli/src/test/java/io/confluent/ksql/cli/console/ConsoleTest.java

```diff
@@ -129,61 +128,529 @@ public void testPrintFinalMessageStreamedRow() throws IOException {
-          new ExecutionPlan("Test Execution Plan")
-      ));
-      console.printKsqlEntityList(entityList);
+    final KsqlEntityList entityList = new KsqlEntityList(ImmutableList.of(
+        new Queries("e", queries)
+    ));
+
+    // When:
+    console.printKsqlEntityList(entityList);
+
+    // Then:
+    final String output = terminal.getOutputString();
+    if (console.getOutputFormat() == OutputFormat.JSON) {
+      assertThat(output, is("[ {\n"
+          + "  \"@type\" : \"queries\",\n"
+          + "  \"statementText\" : \"e\",\n"
+          + "  \"queries\" : [ {\n"
+          + "    \"sinks\" : [ \"Test\" ],\n"
+          + "    \"id\" : \"0\",\n"
+          + "    \"queryString\" : \"select * from t1\"\n"
+          + "  } ]\n"
+          + "} ]\n"));
+    } else {
+      assertThat(output, is("\n"
+          + " Query ID | Kafka Topic | Query String     \n"
+          + "-------------------------------------------\n"
+          + " 0        | Test        | select * from t1 \n"
+          + "-------------------------------------------\n"
+          + "For detailed information on a Query run: EXPLAIN <Query ID>;\n"));
+    }
+  }
+
+  @Test
+  public void testPrintSourceDescription() throws IOException {
+    // Given:
+    final List<FieldInfo> fields = buildTestSchema(
+        Schema.OPTIONAL_BOOLEAN_SCHEMA,
+        Schema.OPTIONAL_INT32_SCHEMA,
+        Schema.OPTIONAL_INT64_SCHEMA,
+        Schema.OPTIONAL_FLOAT64_SCHEMA,
+        Schema.OPTIONAL_STRING_SCHEMA,
+        SchemaBuilder
+            .array(Schema.OPTIONAL_STRING_SCHEMA)
+            .optional()
+            .build(),
+        SchemaBuilder
+            .map(Schema.OPTIONAL_STRING_SCHEMA, Schema.OPTIONAL_INT64_SCHEMA)
+            .optional()
+            .build(),
+        SchemaBuilder
+            .struct()
+            .field("a", Schema.OPTIONAL_FLOAT64_SCHEMA)
+            .optional()
+            .build()
+    );
+
+    final List<RunningQuery> readQueries = ImmutableList.of(
+        new RunningQuery("read query", ImmutableSet.of("sink1"), new EntityQueryId("readId"))
+    );
+    final List<RunningQuery> writeQueries = ImmutableList.of(
+        new RunningQuery("write query", ImmutableSet.of("sink2"), new EntityQueryId("writeId"))
+    );
+
+    final KsqlEntityList entityList = new KsqlEntityList(ImmutableList.of(
+        new SourceDescriptionEntity(
+            "some sql",
+            new SourceDescription(
+                "TestSource", readQueries, writeQueries, fields,
+                DataSourceType.KTABLE.getKsqlType(), "key", "2000-01-01", "stats",
+                "errors", false, "avro", "kadka-topic", 1, 1))
+    ));
+
+    // When:
+    console.printKsqlEntityList(entityList);
+
+    // Then:
+    final String output = terminal.getOutputString();
+    if (console.getOutputFormat() == OutputFormat.JSON) {
+      assertThat(output, is("[ {\n"
```

Comment:
These really big string comparison tests are a bit of a pain for maintainability if we change anything. It's probably better to compare it with a JsonObject that we build instead of just a string so that if we change one field it's clear what comparison failed.

Reply:
Agreed. But this wasn't the primary focus of my quick targeted PR. This is better than what was there, (no testing of output), and we can improve in future PRs.

Reply:
not a blocker :)